### ISSUE TITLE
App crashes on android device when running localhost project

### ISSUE THREAD

**Issue by rok-tel** [expo/expo#4396]
## 🐛 Bug Report
App crashes on android device (not emulator) when running localhost project
### Environment
expo app version on device 2.11.0
expo version on project 32.0.2

app target Android 

### Steps to Reproduce
running expo start on project, 
opening expo app on my android device.

### Expected Behavior
app should load, and render the main screen.

### Actual Behavior
bug screen that is described on following image appears immediately. 

![WhatsApp Image 2019-06-04 at 02 28 06](https://user-images.githubusercontent.com/9028422/58841279-cbcac200-8671-11e9-884b-5ac471d6d44f.jpeg)

**Comment by oriharel** [comment]
Same here

**Comment by MathieuCh** [comment]
Hi, yeah last expo client seems broken. i can't even get to that error screen my expo client is crashing on run. 

Using sdk32

**Comment by oriharel** [comment]
Happens also when running a release channel, not only local.

**Comment by cruzach** [comment]
Looks like this is a duplicate of #4382, which is a duplicate of #4371- continuing conversation there (although it seems to resolved, just waiting on Apple submission now)

**Comment by rok-tel** [comment]
> Looks like this is a duplicate of #4382, which is a duplicate of #4371- continuing conversation there (although it seems to resolved, just waiting on Apple submission now)

I see that the change you made was to ios, does it affect android expo client as well?

**Comment by cruzach** [comment]
Ah you’re right, thanks for pointing that out. Reopening!

**Comment by sjchmiela** [comment]
Does this issue also happen on Snack or only on `expo-cli` projects? Are you able to create a demo reproducing the error?

**Comment by MathieuCh** [comment]
@sjchmiela Apparently on both, just went to https://snack.expo.io/ and picked sdk 32, didn't work

**Comment by sjchmiela** [comment]
What you're saying is that you've:
- opened Snack,
- changed SDK version to 32,
- changed `import Constants from 'expo-constants';` to `import { Constants } from 'expo';`
- ran the Snack on the latest Expo Client
- and it crashed?

**Comment by bot19** [comment]
Hi guys, no idea what Snack is, otherwise would try. I also have this problem. Using `expo-cli`. 

My expo version on project is 32.0.0

We have 2 Pixel 2 XLs, both on Android version 9. The one running expo app client version 2.10.4 works, while the one running 2.11.2 doesn't. I read in related issues that even as early as 2.11.0 is failing and 2.10.6 still works.

Is there a fix in the works? There's no way to rollback expo app versions on android right?

If we are using expo to deploy, will this affect the android package generated? Thanks.

**Comment by picklechips** [comment]
> Hi, yeah last expo client seems broken. i can't even get to that error screen my expo client is crashing on run.
> 
> Using sdk32

I am experiencing the same issue.

For me the Expo Client just crashes after finishing downloading the javascript bundle, however one of the other developers on the same project is experiencing the issue reported in the original post.

Using sdk32, expo android app v2.11.2.

If we revert back to v2.10.6 of the expo client then we experience no issues.

**Comment by MonasteryJohn** [comment]
I am also experiencing the issue with the "backgroundColor" error. In my case it happens after navigating to a particular screen within my app. Unfortunately, I have not been able to narrow down exactly what is causing the issue in that screen and it's children, but it seems to be related to our use of animation in this screen.

This is only impacting Android, and builds done through expo work on the device just fine. Can anyone explain how to revert the expo client version while we wait for a fix for this?

**Comment by cruzach** [comment]
@MonasteryJohn here are links to the old Expo client versions:

Android: https://exp-android-apks.s3.amazonaws.com/Exponent-2.10.6.apk

iOS: https://exp-ios-simulator-apps.s3.amazonaws.com/Exponent-2.10.0.tar.gz

**Comment by MonasteryJohn** [comment]
Thanks to @cruzach for providing a link to the old expo client apk. That helped me find the issue for our app. In our case the problem was with setting backgroundColor: null in some of our native-base button styles. changing those to backgroundColor: 'transparent' got the app working with the latest expo client.

**Comment by picklechips** [comment]
Any update on this?

**Comment by cruzach** [comment]
still need a reproducible example, using a [snack](https://snack.expo.io/) would be best

**Comment by picklechips** [comment]
Just resolved this issue for us. We were using a native-base button with the `transparent` prop, which sets `backgroundColor` to `null`.

They recently released a fix which just changes `null` to `transparent` in order to fix it. It would seem the new client doesn't like `backgroundColor: null`.

Here's the related github issue:
https://github.com/GeekyAnts/NativeBase/issues/2673
and resulting PR:
https://github.com/GeekyAnts/NativeBase/pull/2799

**Comment by sjchmiela** [comment]
Given some time has passed since the issue has been posted and it seems it was caused by `NativeBase` library I'll close this issue tentatively. Thank you for helping everyone out find the root cause of the problem!
